qiita_db/support_files/patches/python_patches/47.py

```diff
@@ -0,0 +1,680 @@
+    TRN.add(sql, ['taxa_summary', sum_taxa_cmd_id, ts_atype_id])
+    sum_taxa_cmd_out_id = TRN.execute_fetchlast()
+    TRN.add(sql, ['distance_matrix', bdiv_cmd_id, dm_atype_id])
+    bdiv_cmd_out_id = TRN.execute_fetchlast()
+    TRN.add(sql, ['rarefaction_curves', arare_cmd_id, rc_atype_id])
+    arare_cmd_out_id = TRN.execute_fetchlast()
+    TRN.add(sql, ['rarefied_table', srare_cmd_id, biom_atype_id])
+    srare_cmd_out_id = TRN.execute_fetchlast()
+
+# At this point we are ready to start transferring the data from the old
+# structures to the new structures. Overview of the procedure:
+# Step 1: Add initial set of artifacts up to rarefied table
+# Step 2: Transfer the "analisys jobs" to processing jobs and create
+#         the analysis artifacts
+db_dir = get_db_files_base_dir()
+with TRN:
+    sql = "SELECT * FROM qiita.analysis"
+    TRN.add(sql)
+    analysis_info = TRN.execute_fetchindex()
+
+    # Loop through all the analysis
+    for analysis in analysis_info:
+        # Step 1: Add the inital set of artifacts. An analysis starts with
+        # a set of BIOM artifacts.
+        sql = """SELECT *
+                 FROM qiita.analysis_filepath
+                    JOIN qiita.filepath USING (filepath_id)
+                    JOIN qiita.filepath_type USING (filepath_type_id)
+                WHERE analysis_id = %s AND filepath_type = 'biom'"""
+        TRN.add(sql, [analysis['analysis_id']])
+        analysis_bioms = TRN.execute_fetchindex()
+
+        # Loop through all the biom tables associated with the current analysis
+        # so we can create the initial set of artifacts
+        for biom_data in analysis_bioms:
+            # Get the path of the BIOM table
+            sql = """SELECT filepath, mountpoint
+                     FROM qiita.filepath
+                        JOIN qiita.data_directory USING (data_directory_id)
+                     WHERE filepath_id = %s"""
+            TRN.add(sql, [biom_data['filepath_id']])
+            # Magic number 0: There is only a single row in the query result
+            fp_info = TRN.execute_fetchindex()[0]
+            filepath = join(db_dir, fp_info['mountpoint'], fp_info['filepath'])
+
+            # We need to check if the BIOM table has been rarefied or not
+            table = load_table(filepath)
+            depths = set(table.sum(axis='sample'))
+            if len(depths) == 1:
+                # The BIOM table was rarefied
+                # Create the initial unrarefied artifact
+                initial_biom_artifact_id = create_non_rarefied_biom_artifact(
+                    analysis, biom_data, table)
+                # Create the rarefaction job
+                rarefaction_job_id, params = create_rarefaction_job(
+                    depths.pop(), initial_biom_artifact_id, analysis,
+                    srare_cmd_id)
+                # Create the rarefied artifact
+                rarefied_biom_artifact_id = create_rarefied_biom_artifact(
```

Comment:
Good question - the name of the function may be misleading. The function is gathering the biom file that is attached to the analysis and creating an artifact with that file (so no new BIOM is created).